server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.given;
+
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/**
+ * Abstract implementation of {@link ContextAwareTest} that suits for tests with mocked context.
+ *
+ * <p>This implementation clears all models before and after all tests.
+ * {@see ModelTests#dropAllModels()}
+ */
+public abstract class TestWithMockContext extends ContextAwareTest {
```

Comment:
What if we name it `FreshContextTest`? And describe it as follows:
```java
An abstract base for {@link io.spine.testing.server.blackbox.BlackBoxContext 
BlackBoxContext}-based tests, which may assemble the bounded-context-under-tests
with different entities, and therefore require to clear the cached knowledge
about {@linkplain io.spine.server.model.Model Bounded Context models}.

<p>This implementation clears all models before and after all tests.

@see ModelTests#dropAllModels()
```

@Artem-Semenov-dev @Vladyslav-Kuksiuk WDYT?

Reply:
It is definitely better.